src/gsy_e/gsy_e_core/sim_results/endpoint_buffer.py

```diff
@@ -403,7 +403,7 @@ def _populate_core_stats_and_sim_state(self, area: "Area"):
             else:
                 if area.parent.current_market is not None:
                     core_stats_dict["energy_rate"] = (
-                        area.strategy.energy_rate.get(area.current_time_slot, None))
+                        area.strategy.energy_rate.get(area.now, None))
```

Comment:
Hmm, not sure whether this will work, because `area.now` also contains the tick duration. How about using `self.current_market_time_slot` ?

Reply:
Sorry, only see the other changes now. Will remove this comment.